anserini_evaluation/README.md

```diff
@@ -0,0 +1,156 @@
+| [splade_distil_CoCodenser_small](http://download-de.europe.naverlabs.com/Splade_Release_Jan22/splade_distil_CoCodenser_small.tar.gz)                    | 37.5            | 97.5 | 71.0      | 46.4           | 48.9        | **0.42**  | 2.0             | 83               |
+
+
+**Note on latency:**: QPS values are computed on a Intel(R) Xeon(R) Gold 6338 CPU @ 2.00GHz, with 128 threads. We simply take the total time (lets call it `t`) for query generation (step 2) on that CPU + the total time for Anserini retrieval. We then compute QPS= |Q|/t, where |Q| is the number of queries in the MS MARCO devset (6980).
+
+**Note on BEIR evaluation**: The BEIR dataset is composed of 18 datasets, of which 4 are not directly accessible (BioASQ, Signal1M, TREC-NEWS, Robust04), and 1 has a multiple-faceted evaluation (CQAdupstack). Removing these five datasets, leads us to the "13 datasets" category; some papers also remove NQ and scidocs, leading to the "11 datasets" category.
+
+The models not included in the repo can be downloaded and extracted as follows (example splade_distil_CoCodenser_small):
+
+```
+SPLADE_URL=http://download-de.europe.naverlabs.com/Splade_Release_Jan22/splade_distil_CoCodenser_small.tar.gz
+SPLADE_NAME=splade_distil_CoCodenser_small
+PATH_SPLADE_WEIGHTS=weights/
+curl -s $SPLADE_URL | tar xvz -C $PATH_SPLADE_WEIGHTS
+```
+
+We then consider that the path to the SPLADE weights is set to the variable $PATH_SPLADE, continuing the example:
+
+```
+PATH_SPLADE=$PATH_SPLADE_WEIGHTS/$SPLADE_NAME
+```
+
+## 1. Generating the document (jsonl) collection
+
+Now that we have everything setup we may start. Considering everything respects the setup, you may generate the document collection via:
+
+```
+SPLIT=16
+INDEX_BATCH_SIZE=100
+OUTPUT_COLLECTION_PATH=anserini_collection/$SPLADE_NAME
+python create_anserini_collection.py --splade_weights_path $PATH_SPLADE --input_collection_path $PATH_MSMARCO/collection.tsv --output_collection_path $OUTPUT_COLLECTION_PATH --index_batch_size $INDEX_BATCH_SIZE --split $SPLIT
+```
+
+Where $SPLIT is the number of splits for the collection jsonl (to allow for multi-threaded indexing), $INDEX_BATCH_SIZE is the batch_size for indexing and $OUTPUT_COLLECTION_PATH. It takes around 16h on a T4 GPU to encode the entire MS MARCO collection.
+
+## 2. Generating the topics (query) files
+
+```
+TOPIC_BATCH_SIZE=100
+TOPIC_NAME=devset_msmarco
+OUTPUT_TOPIC_PATH=topics/$SPLADE_NAME
+python create_anserini_topic.py --splade_weights_path $PATH_SPLADE --input_topic_path $PATH_MSMARCO/queries.dev.small.tsv --output_topic_path $OUTPUT_TOPIC_PATH --topic_batch_size $TOPIC_BATCH_SIZE --output_topic_name $TOPIC_NAME
+```
+
+Where $SPLIT is the number of splits for the collection jsonl (to allow for multi-threaded indexing), $INDEX_BATCH_SIZE is the batch_size for indexing and $OUTPUT_COLLECTION_PATH. If we want to measure latency, 
```

Comment:
I think it lacks one part of a sentence here

Reply:
Corrected the sentence